tools/docs/config.ts

```diff
@@ -233,10 +237,65 @@ export async function generateConfig(dist: string, bot = false): Promise<void> {
         `\n${option.description}\n\n` +
         genTable(Object.entries(el), option.type, option.default);
 
+      if (el.advancedUse) {
+        if (option.name === 'excludeCommitPaths') {
+          configOptionsRaw[headerIndex] += generateAdvancedUse();
+        } else {
+          configOptionsRaw[footerIndex] += generateAdvancedUse();
+        }
+      }
+
+      if (el.followTagAdvancedFunctionality) {
+        configOptionsRaw[headerIndex] +=
+          generateFollowTagAdvancedFunctionality();
+      }
+
+      if (el.groupAdvancedFunctionality) {
+        configOptionsRaw[headerIndex] += generateGroupAdvancedFunctionality();
+      }
+
+      if (el.advancedConfig) {
+        configOptionsRaw[footerIndex] += generateAdvancedConfig();
+      }
+
+      if (el.directEditOfCommitMessageDeprecation) {
+        configOptionsRaw[footerIndex] += generateCommitMsg();
+      }
+
       if (el.experimental) {
         configOptionsRaw[footerIndex] += genExperimentalMsg(el);
       }
     });
 
   await updateFile(`${dist}/${configFile}`, configOptionsRaw.join('\n'));
 }
+
+function generateAdvancedUse(): string {
+  return (
+    '\n<!-- prettier-ignore -->\n!!! warning\n' +
+    '    For advanced use only! Use at your own risk!\n'
+  );
+}
+
+function generateFollowTagAdvancedFunctionality(): string {
+  return "\n<!-- prettier-ignore -->\n!!! warning\n    Advanced functionality.\n    Only use this if you're sure you know what you're doing.\n";
+}
+
+function generateGroupAdvancedFunctionality(): string {
+  return "\n<!-- prettier-ignore --> \n!!! warning\n    Advanced functionality only.\n    Do not use unless you know what you're doing.\n";
+}
+
+function generateAdvancedConfig(): string {
+  return (
+    '\n<!-- prettier-ignore -->\n!!! warning\n' +
+    '    Advanced config, use at your own risk.\n'
+  );
+}
+
+function generateCommitMsg(): string {
+  return (
+    '\n<!-- prettier-ignore -->\n!!! warning\n' +
+    '    Editing of `commitMessage` directly is now deprecated and not recommended.\n' +
+    '    Please instead edit the fields such as `commitMessageAction`, `commitMessageExtra`, etc.\n'
```

Comment:
```suggestion
    '    We deprecated editing the `commitMessage` directly, and we recommend you stop using this config option.\n' +
    '    Instead use config options like `commitMessageAction`, `commitMessageExtra`, and so on, to create the commit message you want.\n'
```

How about this wording? We should explain that _we_ deprecated this feature, and what _we_ recommend. Plus we can rewrite the sentences a bit for clarity as well. 😄